designs/multi-cluster.md

```diff
@@ -50,6 +56,7 @@ This change is important because:
 
 ### Goals
 
+- Provide an interface for plugging in a "cluster provider", which provides a dynamic set of clusters that should be reconciled by registered controllers.
```

Comment:
Not sure this is at the right location. It also overlaps with the goals below.

Would remove 
> - Allow 3rd-parties to plug in their multi-cluster adapter (in source code) into an existing multi-cluster-compatible code-base.

and instead put directly below  `Provide a way to natively write controllers that ...`:
```
Define a cluster provider Go interface that can be implemented by 3rd-parties outside of controller-runtime to adapt an existing multi-cluster-compatible code-base to a concrete environment.
```